sklearn/preprocessing/label.py

```diff
@@ -37,35 +38,66 @@
 ]
 
 
-def _encode_numpy(values, uniques=None, encode=False):
+def _encode_numpy(values, uniques=None, encode=False, encode_missing=False):
     # only used in _encode below, see docstring there for details
+    missing_mask = np.isnan(values)
+    values_masked_nans = values.copy()
+    # mask missing values as existing non-null value
+    values_masked_nans[missing_mask] = values_masked_nans[~missing_mask][0]
```

Comment:
`np.unique` actually handles NaNs (it puts it as the last unique value, which I think is fine for us), so it might be that all of the above is not needed, and that we can directly pass the data to `np.unique`